examples/data-frames.dx

```diff
@@ -284,15 +235,62 @@ def crossJoin {n m f f'} (t:n=>{&...f}) (t':m=>{&...f'}) : (n & m)=>{...f & ...f
 -- It should be sufficient to do the join on a single column. One can always
 -- restructure the table so that the join happens on a tuple of values from
 -- the flattened columns.
-def leftJoin = todo
--- def leftJoin (c:Label) (l:n=>{@c: a & ...f}) (r:m=>{@c: a & ...f'})
---       : n=>{extra: Maybe {...f'} & ...f} = todo
+
+def findInUnordered {a n} [Eq a] (xs:n=>a) (v:a) : Maybe n =
+  -- Wasteful, but at least the parallelism is exposed.
+  (AsList num_found found_table) = argFilter (\x. x == v) xs
+  case num_found == 0 of
+    True -> Nothing
+    False -> Just found_table.(unsafeFromOrdinal _ 0)
+
+-- The implementation below takes (n * m) time, but could be done in O(n log m)
+-- if table r was indexed by a set of values of type a
```

Comment:
What's table r? Perhaps we should try adding a HashMap and doing a hash-based join?

Reply:
I thought `HashMap` was still waiting on something to do with associated types.  In any case, after I wrote this function, I added `argsort`, so now we could also use that approach to improve the time complexity of this function.

Reply:
I think the associated types would be useful for being able to abstract over the concrete map implementation (is it backed by an unordered vector, ordered vector, hash-based vector?), but we should already be able to write a hash map.